src/components/input-dropdown/input-dropdown.component.spec.ts

```diff
@@ -1,77 +1,200 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-
+import { OverlayContainer } from '@angular/cdk/overlay';
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { async, ComponentFixture, inject, TestBed } from '@angular/core/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { RadioButtonModule } from '../radiobutton/radiobutton.module';
 import { InputDropdownComponent } from './input-dropdown.component';
-import { CommonModule } from '@angular/common';
-import { FormsModule } from '@angular/forms';
-import { ElementRef } from '@angular/core';
-import { IconModule } from '../icon';
-import { MenuModule } from '../menu';
+import { InputDropdownModule } from './input-dropdown.module';
+
+describe('InputDropdownComponent', () => {
+    let component: InputDropdownComponent<any>;
+    let fixture: ComponentFixture<InputDropdownComponent<any>>;
+    let overlayContainer: OverlayContainer;
+    let overlayContainerElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                InputDropdownModule,
+                NoopAnimationsModule
+            ]
+        }).overrideComponent(InputDropdownComponent, {
+            set: { changeDetection: ChangeDetectionStrategy.Default }
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(InputDropdownComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should reset filter correctly', () => {
+        component._filterText = 'Tralala';
+
+        spyOn(component.filterChange, 'emit');
+        spyOn(component.filterInputElement.nativeElement, 'focus');
+
+        component.resetFilter(new MouseEvent('click'));
+
+        fixture.detectChanges();
+        expect(component._filterText).toEqual('');
+        expect(component.filterChange.emit).toHaveBeenCalledWith('');
+        expect(component.filterInputElement.nativeElement.focus).toHaveBeenCalledWith();
+    });
+
+    it('should write value', () => {
+        spyOn(component.selectedChange, 'emit');
+        const newValue = 'Bla';
+
+        component.writeValue(newValue);
+
+        expect(component.selected).toEqual(newValue);
+    });
+
+    it('should register onChange', () => {
+        const callbackObject = { callback: () => true };
+        spyOn(callbackObject, 'callback');
+        const newValue = 'Blubb';
+
+        component.registerOnChange(callbackObject.callback);
+        component.onChange(newValue);
```

Comment:
This test isn't really testing anything other than `registerOnChange` function stores the callback in the `component.onChange` variable.

To make this test actually useful, shouldn't call `component.onChange` in our test, as that is us calling it, not the component. We want to check that the component calls it when it should, which would be by making a selection.

The same for the onTouched test below. We should test that actually interacting with the component marks it as touched rather than us telling the component it is touched.